File: util/src/entities/Ban.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { Guild } from "./Guild";
import { User } from "./User";

@Entity("bans")
export class Ban extends BaseClass {
	@RelationId((ban: Ban) => ban.user)
	user_id: string;

	@JoinColumn({ name: "user_id" })
	@ManyToOne(() => User, (user: User) => user.id)
	user: User;

	@RelationId((ban: Ban) => ban.guild)
	guild_id: string;

	@JoinColumn({ name: "guild_id" })
	@ManyToOne(() => Guild, (guild: Guild) => guild.id)
	guild: Guild;

	@RelationId((ban: Ban) => ban.executor)
	executor_id: string;

	@JoinColumn({ name: "executor_id" })
	@ManyToOne(() => User, (user: User) => user.id)
	executor: User;

	@Column()
	ip: string;

	@Column({ nullable: true })
	reason?: string;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Ban.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("bans")]
public partial class Ban
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("user_id")]
    public long? UserId { get; set; }

    [Column("guild_id")]
    public long? GuildId { get; set; }

    [Column("executor_id")]
    public long? ExecutorId { get; set; }

    [Column("ip", TypeName = "character varying")]
    public string? Ip { get; set; }

    [Column("reason", TypeName = "character varying")]
    public string? Reason { get; set; }

    [ForeignKey("ExecutorId")]
    [InverseProperty("BanExecutors")]
    public virtual User? Executor { get; set; }

    [ForeignKey("GuildId")]
    [InverseProperty("Bans")]
    public virtual Guild? Guild { get; set; }

    [ForeignKey("UserId")]
    [InverseProperty("BanUsers")]
    public virtual User? User { get; set; }
}
